File: packages/agent/web/client/src/toolsummary.test.ts
import { readFileSync } from 'node:fs'
import { fileURLToPath } from 'node:url'
import { dirname, join } from 'node:path'
import { describe, expect, it } from 'vitest'
import { summarizeToolNames } from './toolsummary'

// The SAME golden fixtures the Go summarizer test loads
// (packages/tui/testdata/tool_summary_golden.json). Pinning both languages to
// one file means a change to the summary format on either side fails the
// other's test — the TUI and web summaries cannot silently drift.
const here = dirname(fileURLToPath(import.meta.url))
const fixturePath = join(here, '../../../../tui/testdata/tool_summary_golden.json')
const cases = JSON.parse(readFileSync(fixturePath, 'utf8')) as { names: string[]; want: string }[]

describe('summarizeToolNames matches the shared golden fixtures', () => {
  it('loads fixtures', () => {
    expect(cases.length).toBeGreaterThan(0)
  })
  for (const c of cases) {
    it(JSON.stringify(c.names), () => {
      expect(summarizeToolNames(c.names)).toBe(c.want)
    })
  }
})

File: packages/tui/toolsummary_test.go
package tui

import (
	"encoding/json"
	"os"
	"path/filepath"
	"testing"
)

// TestSummarizeToolNamesGolden pins the Go summarizer to the shared golden
// fixtures. The web UI's summarizeToolNames test (toolsummary.test.ts) loads
// the SAME file, so a change to the format on either side breaks the other's
// test — the two implementations cannot silently drift.
func TestSummarizeToolNamesGolden(t *testing.T) {
	raw, err := os.ReadFile(filepath.Join("testdata", "tool_summary_golden.json"))
	if err != nil {
		t.Fatal(err)
	}
	var cases []struct {
		Names []string `json:"names"`
		Want  string   `json:"want"`
	}
	if err := json.Unmarshal(raw, &cases); err != nil {
		t.Fatal(err)
	}
	if len(cases) == 0 {
		t.Fatal("no golden fixtures loaded")
	}
	for _, c := range cases {
		if got := summarizeToolNames(c.Names); got != c.Want {
			t.Errorf("summarizeToolNames(%v) = %q, want %q", c.Names, got, c.Want)
		}
	}
}
